src/ncp/ncp_spinel.cpp

```diff
@@ -72,7 +72,9 @@ void NcpSpinel::Init(ot::Spinel::SpinelDriver &aSpinelDriver, PropsObserver &aOb
 
 void NcpSpinel::Deinit(void)
 {
-    mSpinelDriver = nullptr;
+    mSpinelDriver              = nullptr;
+    mIp6AddressTableCallback   = nullptr;
+    mNetifStateChangedCallback = nullptr;
```

Comment:
Do we need to initialize these variables to `nullptr` in the constructor?

Reply:
The `std::function` has default constructor to let the member be `nullptr`. So I think it's not necessary to initialize them explicitly. Thoughts?